### PR TITLE
Document new XS container size and flexible resources allocation

### PR DESCRIPTION
**Pull request by tucksaun**

### CODE REVIEW

File: cookbooks/resources_allocation.rst [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+   environment is treated like a
+   :ref:`development environment <dev-env-sizing>` and container sizes are set
+   to ``S``.
+
+You can change your project plan at anytime using the
+``symfony project:scale <plan>`` command.
+
+Possible production plan sizes are:
+
+* Standard
```
Comment (fabpot) [inline cookbooks/resources_allocation.rst:27]: When running `symfony project:scale`, the arg is the plan name like l, m, ..., right? If that's the case, we should probably add them to the list below

File: cookbooks/resources_allocation.rst [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+* Medium
+* Large
+* X-Large
+* 2X-Large
+
+Each comes with increasing CPU and memory compared to the previous one. Please
+refer to the `SymfonyCloud pricing page`_ to get accurate resource allocation.
+
+Container size
```
Comment (fabpot) [inline cookbooks/resources_allocation.rst:36]: Size

File: cookbooks/resources_allocation.rst [hunk 3]
```diff
@@ -0,0 +1,176 @@
+Resources allocation
```
Comment (fabpot) [inline cookbooks/resources_allocation.rst:1]: Allocation

File: cookbooks/resources_allocation.rst [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+best option. In particular it will allow the application to grow as the plan
+size increases. If a container is set at a fixed size, however, it will not get
+any bigger and the extra resources might go to waste.
+
+The main use-case where setting size manually is helpful is when setting
+particular containers to a fixed size to indicate that it really doesn't need
+much CPU or RAM or if its workload is fixed and doesn't need to scale much.
+**Static sites**, **background workers**, and other **low-need containers** can
+be set that way to allow more resources to go to other applications and services
+containers that really need them. For example, keep your main web container one
```
Comment (fabpot) [inline cookbooks/resources_allocation.rst:73]: on instead of one?

Comment (fabpot) [reply cookbooks/resources_allocation.rst:73]: or to?

File: cookbooks/resources_allocation.rst [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+* For especially memory intensive per-process applications, setting a higher
+  ``memory_ratio`` (see the `custom resources allocation`_ section) will give
+  those applications more head room to grow. However, be aware that doing so may
+  draw memory away from other applications or services that need it. Try to
+  optimize your application before throwing more resources at it.
+
+In both cases you will need to do your own profiling to determine optimal values
+for your application.
+
+Service sizes
```
Comment (fabpot) [inline cookbooks/resources_allocation.rst:94]: Sizes

File: cookbooks/resources_allocation.rst [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+~~~~~~~~~~~~~
+
+Service sizes are set via the ``size`` property in :doc:`../services/intro`.
+Possible values are ``AUTO`` (the default if not set), ``S``, ``M``, ``L``,
+``XL``, ``2XL`` and ``4XL``.
+
+Application sizes
```
Comment (fabpot) [inline cookbooks/resources_allocation.rst:101]: Sizes

File: cookbooks/resources_allocation.rst [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+Possible production plan sizes are:
+
+* Standard
+* Medium
+* Large
+* X-Large
+* 2X-Large
+
+Each comes with increasing CPU and memory compared to the previous one. Please
+refer to the `SymfonyCloud pricing page`_ to get accurate resource allocation.
```
Comment (fabpot) [inline cookbooks/resources_allocation.rst:34]: But this is listed in a section below.

File: cookbooks/resources_allocation.rst [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+M    1.0  320 MB
+L    2.0  576 MB
+XL   4.0  1088 MB
+2XL  8.0  2112 MB
+==== ==== ===========
+
+.. note::
+
+   The smaller ``XS`` container size is only available when using custom
+   resources allocation as describe bellow in the `custom resources allocation`_
```
Comment (fabpot) [inline cookbooks/resources_allocation.rst:126]: below

File: cookbooks/resources_allocation.rst [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+   section.
+
+Custom resources allocation
```
Comment (fabpot) [inline cookbooks/resources_allocation.rst:129]: Custom Resources Allocation

File: cookbooks/resources_allocation.rst [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+configurable. Two properties can be adjusted:
+
+.. code-block:: yaml
+
+   resources:
+       base_memory: 64 #default and minimum
+       memory_ratio: 256 #defautl, minimum: 128
+
+``base_memory`` represents a fixed amount of memory (in megabytes) allocated to
+the container.  ``memory_ratio`` is an amount of memory allocated per CPU share.
```
Comment (fabpot) [inline cookbooks/resources_allocation.rst:149]: double spaces

File: cookbooks/resources_allocation.rst [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
+For example: an ``S`` container gets a 0.5 CPU share, with the default
+configuration it would get 64 + 0.5 x 256 = 192 MB of RAM.
+If the same container gets its size increased to an XL, it would now get
```
Comment (fabpot) [inline cookbooks/resources_allocation.rst:152]: XL should be quoted

File: cookbooks/workers.rst [hunk 12]
```diff
@@ -56,6 +56,13 @@ specified by the ``start`` key terminates it will be restarted automatically.
     possible. Every data sharing needs to be done using :doc:`services
     </services/intro>`.
 
+.. tip::
+
+   While worker and application containers share the same ``size`` by default,
+   they usually benefit from being tweak separately. Read the documentation
```
Comment (fabpot) [inline cookbooks/workers.rst:62]: tweaked

File: services/intro.rst [hunk 13]
```diff
@@ -94,7 +94,12 @@ generate an error if present.
 Size
 ~~~~
 
-This attribute has the same effect as for :ref:`application containers <config-size>`.
+Tweaks the service container size. Legal values for the size key are ``AUTO``
```
Comment (fabpot) [inline services/intro.rst:97]: Legal -> supported

File: services/intro.rst [hunk 14]
```diff
@@ -94,7 +94,12 @@ generate an error if present.
...
+(the default), ``S``, ``M``, ``L``, ``XL``, ``2XL`` and ``4XL``.
+
+Read the documentation about
+:doc:`resources allocations <../cookbooks/resources_allocation>` for more
+details about the influence of this parameter.
```
Comment (fabpot) [inline services/intro.rst:102]: about the impact of this parameter.

**Review by fabpot — CHANGES_REQUESTED** [review]

File: cookbooks/resources_allocation.rst (hunk 1 above)
Comment (tucksaun) [reply cookbooks/resources_allocation.rst:27]: The optional arg is one:
- `development`/`dev`
- `standard`
- `medium`
- `large`
- `xlarge`
- `2xlarge`

File: cookbooks/resources_allocation.rst (hunk 7 above)
Comment (tucksaun) [reply cookbooks/resources_allocation.rst:34]: bellow is about container size, not plan size.
your comment means we need to clarify this.

what about:
> Each plan comes with increasing CPU and memory compared to the previous one. Please refer to the `SymfonyCloud pricing page`_ to get the official allowance coming with each plan.

?

**Pull request merged**